src/tsk_fs_file_handler.rs

```diff
@@ -0,0 +1,170 @@
+#![allow(non_camel_case_types)]
+
+use std::io::{Read, Seek, SeekFrom};
+use std::ffi::{CStr, CString};
+use super::{
+    tsk_fs_file::TskFsFile,
+    tsk_fs_attr::TskFsAttrType,
+    errors::TskError,
+    bindings as tsk
+};
+
+#[derive(Copy, Clone)]
+pub enum TskFsFileReadFlag {
+    None,
+    Slack,
+    NoID
+}
+
+#[derive(Copy, Clone)]
+pub struct TskFsFileHandler<'fs>{
```

Comment:
Lets call this `TskFsFileHandle` as its more generic and represents a File Handle. `TskFsFileHandler` implies more management type operations.

I think it will be easier for `TskFsFileHandle` to store an owned TskFsAttr. That is because both `attr_to_read` and `id`, and `attr_size` are all stored in the `TskFsAttr`. This will make for a cleaner implementation.

This will also make it easier to get `TskFsFileHandle`s from attributes, including a file's default attribute which makes things easy.

There are pieces of this code that are NTFS specific and would not work with other file systems.

Reply:
I agree and done 👍